File: TrainingTracker/backend/src/TrainingTracker.Domain/Entities/Course.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TrainingTracker.Domain.Entities;

public class Course : BaseEntity
{
    [Required]
    [MaxLength(300)]
    public string Title { get; set; } = null!;

    [MaxLength(2000)]
    public string? Description { get; set; }

    // Optional: reference to a category
    public Guid? CourseCategoryId { get; set; }
    public CourseCategory? CourseCategory { get; set; }

    // Estimated duration in hours
    public int? EstimatedHours { get; set; }

    // Navigation: assignments for this course
    public ICollection<CourseAssignment> CourseAssignments { get; set; } = new List<CourseAssignment>();
}

File: TrainingTracker/frontend/training-tracker-ui/src/app/shared/models/course.ts
export interface Course {
  id: string;
  title: string;
  description?: string;
  courseCategoryId?: string;
  courseCategoryName?: string;
  estimatedHours?: number;
}

export interface CreateCourse {
  title: string;
  description?: string;
  courseCategoryId?: string;
  estimatedHours?: number;
}

export interface UpdateCourse extends CreateCourse {
  id: string;
}

export interface CourseCategory {
  id?: string;
  name: string;
}
